submission-via-api/submit.py

```diff
@@ -0,0 +1,43 @@
+import requests, json
+
+def submit(submitterEmail,secret,key,submission_part,all_parts, data):
+        submission = {
+                    "assignmentKey": key,
+                    "submitterEmail":  submitterEmail,
+                    "secret":  secret,
+                    "parts": {}
+                  }
+        for part in all_parts:
+            if part == submission_part:
+                submission["parts"][part] = {"output": data}
+            else:
+                submission["parts"][part] = dict()
+
+        response = requests.post('https://www.coursera.org/api/onDemandProgrammingScriptSubmissions.v1', data=json.dumps(submission))
+
+        if response.status_code == 201:
+            print ("Submission successful, please check on the coursera grader page for the status")
+        else:
+            print ("Something went wrong, please have a look at the reponse of the grader")
+            print ("-------------------------")
+            print (response.text)
+            print ("-------------------------")
+
+
+
+key = "Tdma9jdAEead7w5LqXvQQQ" # Assessment key found on the authoring side (top of page)
```

Comment:
I hope this key doesn't correspond to anything real assessment?

Reply:
@sashi-coursera Nope! Key and part IDs are fake